### PR TITLE
port View to rst

### PR DESCRIPTION
**Pull request by fnrizzi**
co-authors: Marcin Wrobel, Antoine Meyer

This PR superseeds https://github.com/kokkos/kokkos-core-wiki/pull/145 

before (.md) | after (.rst)
:----------------: | :-------------:
image | image
image | image 
---- | image
image | image
image | image
image | image
image | image
image | image
image | image
image | image
image | image

### CODE REVIEW

File: docs/source/API/core/view/view.rst [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+
+.. code-block:: cpp
+
+    template <class DataType [, class LayoutType] [, class MemorySpace] [, class MemoryTraits]>
+    class View;
+
+Parameters
+~~~~~~~~~~
+
+.. _LayoutRight: layoutRight.html
```
Comment (nmm0) [inline docs/source/API/core/view/view.rst:26]: Just a note that we will have to replace these with cross-references once everything is ported over

**Review by nmm0 — APPROVED** [review]
Looks good to me!

File: docs/source/API/core/view/view.rst [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
+    DstType dst_view(src_view);
+    dst_view = src_view;
+
+The following conditions must be met at and are evaluated at compile time:
+
+* ``DstType::rank == SrcType::rank``
+* ``DstType::non_const_value_type`` is the same as ``SrcType::non_const_value_type``
+* If ``std::is_const<SrcType::value_type>::value == true`` than ``std::is_const<DstType::value_type>::value == true``.
+* ``MemorySpaceAccess<DstType::memory_space,SrcType::memory_space>::assignable == true``
+* If ``DstType::dynamic_rank != DstType::rank`` and ``SrcType::dynamic_rank != SrcType::rank`` than for each dimension ``k`` which is compile time for both it must be true that ``dst_view.extent(k) == src_view.extent(k)``
```
Comment (ajpowelsnl) [inline docs/source/API/core/view/view.rst:507]: Typo:  "than" should be then

File: docs/source/API/core/view/view.rst [hunk 3]
```diff
@@ -0,0 +1,577 @@
...
+
+Additionally the following conditions must be met at runtime:
+
+* If ``DstType::dynamic_rank != DstType::rank`` than for each compile time dimension ``k`` it must be true that ``dst_view.extent(k) == src_view.extent(k)``.
```
Comment (ajpowelsnl) [inline docs/source/API/core/view/view.rst:511]: Typo:  "than" should be then

File: docs/source/API/core/view/view.rst [hunk 4]
```diff
@@ -0,0 +1,577 @@
...
+
+Furthermore there are rules which must be met if ``DstType::array_layout`` is not the same as ``SrcType::array_layout``.
+These rules only cover cases where both layouts are one of ``LayoutLeft``, ``LayoutRight`` or ``LayoutStride``
+
+* If neither ``DstType::array_layout`` nor ``SrcType::array_layout`` is ``LayoutStride``:
+    - If ``DstType::rank > 1`` than ``DstType::array_layout`` must be the same as ``SrcType::array_layout``.
```
Comment (ajpowelsnl) [inline docs/source/API/core/view/view.rst:517]: Typo: "than" should be then

**Review by ajpowelsnl — CHANGES_REQUESTED** [review]
There are a few obvious (spelling) errors that should be corrected.  If possible, rework the sentence that doesn't make much sense.  I can help with that.

**Review by ajpowelsnl — CHANGES_REQUESTED** [review]
On line 82, the sentence, "Enums can be combined bit combined" doesn't make much sense. Can we fix obvious problems in the main porting task (.md --> .rst)? Assuming so, @crtrott , @nmm0, @dalg24, is the intended meaning of this sentence something along the lines of, "Enums parameters for the MemoryTraits class hold bitwise combinations of flags" ...?

File: docs/source/API/core/view/view.rst [hunk 5]
```diff
@@ -0,0 +1,577 @@
...
+
+.. _Atomic: ../atomics.html
+
+.. |Atomic| replace:: :cppkokkos:func:`Atomic`
+
+
+* ``MemoryTraits``:
+
+  Sets access properties via enum parameters for the templated ``Kokkos::MemoryTraits<>`` class.
+  Enums can be combined bit combined. Posible values:
```
[the remark this answers is not in the document]
Comment (masterleinad) [reply docs/source/API/core/view/view.rst:82]: Yes, something like that. Maybe, say "Possible template parameters are bit-combinations of the following flags:"

**Comment by fnrizzi** [comment]
I am happy to edit obvious typos

File: docs/source/API/core/view/view.rst (hunk 1 above)
Comment (fnrizzi) [reply docs/source/API/core/view/view.rst:26]: yes

**Review by crtrott — APPROVED** [review]
All in all looks good. But yeah fixing that sentence there might be good.

**Review by ajpowelsnl — APPROVED** [review]

**Pull request merged**